Packet++/src/Asn1Codec.cpp

```diff
@@ -771,4 +776,172 @@ namespace pcpp
 		m_ValueLength = 0;
 		m_TotalLength = 2;
 	}
+
+	Asn1ObjectIdentifier::Asn1ObjectIdentifier(const uint8_t* data, size_t dataLen)
+	{
+		if (!data || dataLen == 0)
+		{
+			throw std::invalid_argument("Malformed OID: Not enough bytes for the first component");
+		}
+
+		size_t currentByteIndex = 0;
+		std::vector<uint32_t> components;
+
+		uint8_t firstByte = data[currentByteIndex++];
+		components.push_back(static_cast<uint32_t>(firstByte / 40));
```

Comment:
Consider explaining what the 40 magic number is in the comments.

Reply:
Added documentation here: https://github.com/seladb/PcapPlusPlus/pull/1849/commits/d69094760b23411f7664650b07a4d0c2c50ec66a